addons/purchase_stock/models/purchase.py

```diff
@@ -347,11 +347,39 @@ def _prepare_stock_moves(self, picking):
         res = []
         if self.product_id.type not in ['product', 'consu']:
             return res
+
         qty = 0.0
         price_unit = self._get_stock_move_price_unit()
         for move in self.move_ids.filtered(lambda x: x.state != 'cancel' and not x.location_dest_id.usage == "supplier"):
             qty += move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom, rounding_method='HALF-UP')
-        template = {
+
+        move_dests = self.move_dest_ids
+        if not move_dests:
+            move_dests = self.move_ids.move_dest_ids
+
+        if not move_dests:
+            qty_to_attach = 0
+            qty_to_push = self.product_qty - qty
+        else:
+            move_dests_initial_demand = self.product_id.uom_id._compute_quantity(
+                sum(move_dests.filtered(lambda m: m.state != 'cancel').mapped('product_qty')),
```

Comment:
Not sure it works with returns

Reply:
can you detail the flow you're thinking of?

Reply:
confirm po + do a return  + change qty